dev/planning-example.py

```diff
@@ -0,0 +1,206 @@
+obs.sunset() # Will be use time=Time.now(), which='nearest'
+
+# sunrise
+obs.sunrise(time=time_obs, which='nearest') # Default
+obs.sunrise(time=time_obs, which='next')
+obs.sunrise(time=time_obs, which='previous')
+
+# moon rise
+obs.moonrise(time=time_obs, which='nearest')
+
+# moon set
+obs.moonset(time=time_obs, which='nearest')
+
+# The above functions can be called with an `angle` keyword argument to specify
+# a particular horizon angle for rising or setting, or can be called with
+# convenience functions for particular morning/evening twilight.
+# For example, to compute astronomical twilight by specifying the `angle`:
+obs.sunset(time=time_obs, which='next', angle=18*u.degree) 
+
+# evening (astronomical) twilight
+obs.evening_astronomical(time=time_obs)
+
+# evening (nautical) twilight
+obs.evening_nautical(time=time_obs)
+
+# evening (civil) twilight
+obs.evening_civil(time=time_obs)
+
+# morning (nautical) twilight
+obs.morning_nautical(time=time_obs)
+
+# morning (civil) twilight
+obs.morning_civil(time=time_obs)
+
+# morning (astronomical) twilight
+obs.morning_astronomical(time=time_obs)
+
+# what is the moon illumination?
+# returns a float, which is percentage of the moon illuminated
+obs.moon_illumination(time=time_obs)
+
+# what is the moon altitude and azimuth?
+obs.moon_altaz(time=time_obs)
+
+# Other sun-related convenience functions:
+obs.noon(time=time_obs, which='nearest')
+obs.midnight(time=time_obs, which='nearest')
+
+# ==============
+# Target objects
+# ==============
+'''
+A target object defines a single observation target, with coordinates
+and an optional name.
+'''
+from astroplan import FixedTarget, airmass_plot
+
+# Define a target.
+
+from astropy.coordinates import SkyCoord
+t1 = FixedTarget(name='Polaris',
+                 coord=SkyCoord('02h31m49.09s', '+89d15m50.8s', frame='icrs'))
+
+# Leaves scope for NonSiderealTarget, etc.
+
+# Convenience methods for looking up/constructing targets by name via
+# astroquery:
+t1 = FixedTarget.from_name('Polaris')
+
+# ================================
+# Condition objects, observability
+# ================================
+
+"""
+Q: Can I observe this target on the night of May 1, 2015 between 18:30
+and 05:30 HST, above airmass 1.2, on a telescope that can observe between 
+15 degrees and 89 degrees altitude?
+"""
+
+# first we define the conditions
+from astroplan import TimeWindow, AltitudeWindow, AboveAirmass, is_observable
+
+# times can be passed in as strings (interpreted as for get_date()) or
+# as astropy Time or datetime objects
+time1, time2 = Time("2015-05-01 18:30"), Time("2015-05-02 05:30")
+constraint_list = [TimeWindow(time1, time2), AboveAirmass(1.2),
+                   AltitudeWindow(15.0*u.deg, 89.0*u.deg)]
+# (AboveAirmass will be a subclass of AltitudeWindow)
+
+# Combine a list of constraints to run on Observer, FixedTarget, and time to 
+# determine the observability of target
+constraints = is_observable(constraint_list, obs, t1, time_obs)
```

Comment:
Can you leave a comment here that `is_observable` is a "provisional" name and it  should probably end up a method of something?  I'm fine with the choice of kicking the decision down the road.  But I want to make sure we don't end up with an API that makes it harder to work fast (which could happen with this as a function instead of a method because it doesn't have a place to cache anything.)